src/map/battle.cpp

```diff
@@ -2375,21 +2375,34 @@ static bool is_attack_left_handed(struct block_list *src, int skill_id)
  */
 static bool is_attack_critical(struct Damage* wd, struct block_list *src, struct block_list *target, int skill_id, int skill_lv, bool first_call)
 {
-	struct status_data *sstatus = status_get_status_data(src);
-	struct status_data *tstatus = status_get_status_data(target);
-	struct status_change *sc = status_get_sc(src);
-	struct status_change *tsc = status_get_sc(target);
-	struct map_session_data *sd = BL_CAST(BL_PC, src);
-	struct map_session_data *tsd = BL_CAST(BL_PC, target);
-
 	if (!first_call)
 		return (wd->type == DMG_CRITICAL || wd->type == DMG_MULTI_HIT_CRITICAL);
 
+#ifdef RENEWAL
 	if (skill_id == NPC_CRITICALSLASH || skill_id == LG_PINPOINTATTACK) //Always critical skills
```

Comment:
Should we just de-hardcode these and put them into the renewal skill_db?

Reply:
i do think moving it under the `` if( skill_id && !skill_get_nk(skill_id,NK_CRITICAL) ) `` to let the skill_db decide if it can be  crit or not is better

Reply:
Right. Simply removing this check and then just adding the flags to the db should be fine. The check right below this one will catch it.

Reply:
i already added the skill_db part , however i don't think we can remove the check as those skills have 100% chance to crit , if we remove the check they wont be 100% , and there is no way to put  x% in the skill_db <details> <summary>maybe x% in the db is a good idea</summary>(it's a good idea to do in my opinion but maybe in another pr ? instead of Crecital: true we can make it Cricital: chanse , but i don't thing there is a lot of skills that would benefit from this) </details>
so i think the best is to just move the check down

Reply:
Ah, you're right. I forgot about the rate. Just leave the hardcoded parts. We can come up with something later on, it won't be related to this PR.